### PR TITLE
MRG: Add _check_stc

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This small PR refactors `_triage_stc` and adds `_check_stc` which is useful for #8335.

### CODE REVIEW

File: mne/viz/_brain/_brain.py [hunk 1]
```diff
@@ -1493,6 +1493,10 @@ def add_data(self, array, fmin=None, fmid=None, fmax=None,
                                  f'{time_label_size}')
 
         hemi = self._check_hemi(hemi, extras=['vol'])
+        try:
+            stc, array, vertices = self._check_stc(hemi, array, vertices)
+        except ValueError:
+            return  # No data to add, skipping
```
Comment (larsoner) [inline mne/viz/_brain/_brain.py:1499]: This seems weird -- someone calls `add_data` but no data gets added? Shouldn't this be an error? What is the use case?

File: mne/viz/_brain/_brain.py [hunk 2]
```diff
@@ -2893,6 +2930,8 @@ def scale_data_colormap(self, fmin, fmid, fmax, transparent,
         %(verbose_meth)s
         """
         # XXX fmid/transparent/center not used, this is a bug
+        if 'ctable' not in self._data:
+            return
```
Comment (larsoner) [inline mne/viz/_brain/_brain.py:2934]: There is no colormap but someone calls `scale_data_colormap` -- same as above, seems like it should be an error to do this

**Comment by GuillaumeFavelier** [comment]
The failures of `codespell` don't seem related:

```
Error: mne/channels/tests/test_montage.py:1101: worflow ==> workflow
Error: mne/gui/_kit2fiff_gui.py:516: Trough ==> Through
Error: mne/gui/help/kit2fiff.json:7: Trough ==> Through
Error: doc/install/contributing.rst:771: Vizualization ==> Visualization
```

But 2 of them seem legit:

<details>

```diff
diff --git a/doc/install/contributing.rst b/doc/install/contributing.rst
index 350b8ed47..9cb15b8f1 100644
--- a/doc/install/contributing.rst
+++ b/doc/install/contributing.rst
@@ -768,7 +768,7 @@ for the return value, even if the function or method's return value is never
 stored under that name in the code.
 
 
-Vizualization
+Visualization
 -------------
 
 Visualization capabilities should be made available in both function and method
diff --git a/mne/channels/tests/test_montage.py b/mne/channels/tests/test_montage.py
index f7460c5d2..b8660db2b 100644
--- a/mne/channels/tests/test_montage.py
+++ b/mne/channels/tests/test_montage.py
@@ -1098,7 +1098,7 @@ def test_transform_to_head_and_compute_dev_head_t():
 
     montage_meg = make_dig_montage(hpi=hpi_dev, coord_frame='meg')
 
-    # Test regular worflow to get dev_head_t
+    # Test regular workflow to get dev_head_t
     montage = montage_polhemus + montage_meg
     fids, _ = _get_fid_coords(montage.dig)
     for kk in fids:
```

</details>

Strangely, `master` (92a29ff3c4e95fdae546cafd193cd0dc899a43bd) does not have this issue :thinking:

**Comment by agramfort** [comment]
push the fix in this PR it's fine

>

**Comment by GuillaumeFavelier** [comment]
[Circle fails](https://app.circleci.com/pipelines/github/mne-tools/mne-python/5025/workflows/a5692e7f-7862-411d-84fa-6b7655e4657a/jobs/23804/parallel-runs/0/steps/0-111?invite=true) at the `Check PyQt5` step, is it because of the release of `PyQt5 5.15.2`?

image

**Comment by agramfort** [comment]
pin the version to try

>

File: mne/viz/_3d.py [hunk 3]
```diff
@@ -1919,22 +1897,31 @@ def _plot_stc(stc, subject, surface, hemi, colormap, time_label,
...
             kwargs["clim"] = clim
-            kwargs["volume_options"] = volume_options
-            kwargs["src"] = src_vol
+            kwargs["src"] = src
         kwargs.update({} if add_data_kwargs is None else add_data_kwargs)
+        if volume and src.kind == 'volume':
+            continue
         with warnings.catch_warnings(record=True):  # traits warnings
             brain.add_data(**kwargs)
         brain.scale_data_colormap(fmin=scale_pts[0], fmid=scale_pts[1],
```
Comment (larsoner) [inline mne/viz/_3d.py:1917]: Shouldn't this also be outside the loop and probably after the new `volume` lines?

Comment (GuillaumeFavelier) [reply mne/viz/_3d.py:1917]: This skips the case where we don't want to add surface when there is none similarly to https://github.com/mne-tools/mne-python/blob/2b4605fbf50d1d03edc34ad013b5b4577f948b81/mne/viz/_3d.py#L1901-L1902

Comment (GuillaumeFavelier) [reply mne/viz/_3d.py:1917]: Related to https://github.com/mne-tools/mne-python/pull/8547#discussion_r527768391 and https://github.com/mne-tools/mne-python/pull/8547#discussion_r527769357:

> someone calls add_data but no data gets added?

**Comment by larsoner** [comment]
Thanks @GuillaumeFavelier

**Comment by larsoner** [comment]
@GuillaumeFavelier this broke CircleCI:
```
Traceback (most recent call last):
  File "/home/circleci/project/tutorials/misc/plot_ecog.py", line 195, in <module>
    clim = dict(kind='value', lims=[vmin * 0.9, vmin, vmax])
  File "/home/circleci/project/mne/source_estimate.py", line 653, in plot
    brain = plot_source_estimates(
  File "<decorator-gen-139>", line 24, in plot_source_estimates
  File "/home/circleci/project/mne/viz/_3d.py", line 1794, in plot_source_estimates
    return _plot_stc(
  File "/home/circleci/project/mne/viz/_3d.py", line 1973, in _plot_stc
    brain.setup_time_viewer(time_viewer=time_viewer,
  File "/home/circleci/project/mne/viz/_brain/_brain.py", line 419, in setup_time_viewer
    self._configure_point_picking()
  File "/home/circleci/project/mne/viz/_brain/_brain.py", line 920, in _configure_point_picking
    ind = np.unravel_index(np.argmax(np.abs(use_data), axis=None),
  File "<__array_function__ internals>", line 5, in argmax
  File "/home/circleci/.local/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 1188, in argmax
    return _wrapfunc(a, 'argmax', axis=axis, out=out)
  File "/home/circleci/.local/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 58, in _wrapfunc
    return bound(*args, **kwds)
ValueError: attempt to get argmax of an empty sequence
```
Can you look?

**Pull request merged**